src/store-api/src/region_engine.rs

```diff
@@ -120,6 +124,65 @@ impl From<PbRegionRole> for RegionRole {
     }
 }
 
+/// Output partition properties of the [RegionScanner].
+#[derive(Debug)]
+pub enum ScannerPartitioning {
+    /// Unknown partitioning scheme with a known number of partitions
+    Unknown(usize),
+}
+
+impl ScannerPartitioning {
+    /// Returns the number of partitions.
+    pub fn num_partitions(&self) -> usize {
+        match self {
+            ScannerPartitioning::Unknown(num_partitions) => *num_partitions,
+        }
+    }
+
+    /// Converts to datafusion's [Partitioning].
+    pub fn to_df_partitioning(&self) -> Partitioning {
+        match self {
+            ScannerPartitioning::Unknown(num_partitions) => {
+                Partitioning::UnknownPartitioning(*num_partitions)
+            }
+        }
+    }
```

Comment:
We suggest not to provide such a method. Leaving `ScannerPartitioning` internal (in this project) only. DF's `Partitioning` is 
- for generic data partition, especially intermediate data
- hard to add new variants
- not as expressive as we want

And the `ScannerPartitioning` is likely to be used in our optimizer rule, which also bypasses the `Partitioning` interface

Reply:
We always map our partitions to `unknown` partitioning in DF. The main purpose of this method is to tell df that we can return multiple streams. I can change this to `partition_count()`.

Reply:
Removed in https://github.com/GreptimeTeam/greptimedb/pull/3948/commits/537f4b5f47934bf77b47bf499b81100f12f7b42d

Reply:
>We always map our partitions to unknown partitioning in DF.

Makes a little more sense to me. Let's keep it for now and review the usage one month later

Reply:
Already removed. Now only one place uses it.